scripts/common/addon.sh

```diff
@@ -168,7 +177,15 @@ function addon_cleanup() {
 
 function addon_has_been_applied() {
     local name=$1
-    last_applied=$(kubectl get configmap -n kurl kurl-last-config -o jsonpath="{.data.addons-$name}")
+
+    if [ "$name" = "containerd" ]; then
+        if [ -f $DIR/containerd-last-applied ]; then
```

Comment:
is there any reason why you chose to do this rather than say add to the config map at some point after k8s is installed, say in the outro?

Reply:
when you say `Kubernetes cluster may not be up`, do you mean that on initial install or may it not be up on upgrade as well?

Reply:
I meant that it won't be up on initial install.

Using the config map doesn't work for containerd because remote nodes may be workers and not have kubectl. And we have to keep track of the version of containerd installed on that node. It's not like other addons where we only care about the last version applied to the cluster.